libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1404,6 +1406,7 @@ string ABIFunctions::calldataAccessFunction(Type const& _type)
 				w("handleLength", "");
 				w("return", "value");
 			}
+			// TODO this makes a difference!
```

Comment:
@ekpyron can you check if the length check is still correct when this returns 32 for dynamically encoded types?